magento/pwa-studio
# PWA-1712: [bug]: fix region showing as number on braintree summary

## Root Cause
The BraintreeSummary component was using region.code as the string value for State, this was causing a bug on countries like France where region.code is shown as a number, which is not useful for the client, in this case the right value to be used is region.label from [CartAdressRegion](https://devdocs.magento.com/guides/v2.4/graphql/queries/cart.html#CartAddressRegion)  

## Description
This PR changes the string value used for the state field on BraintreeSummary address.

## Related Issue







Closes #3322 .

## Acceptance



On CheckoutPage the Payment Information should not show any region code. It should show the region name (ex: Ardèche) as it is on Shipping Information.

### Verification Stakeholders



### Specification




## Verification Steps





1. Add product to cart and go to checkout form
2. Enter shipping address with France as country and select shipping method
3. Enter payment method, uncheck billing is same as shipping and enter France country but different address
4. Click on Review order button
5. Now observe the region in billing address

#### Test scenario(s) for direct fix/feature




1. Verify that the region code is no longer showing on billing address, instead it should show region label (name).

#### Test scenario(s) for any existing impacted features/areas





#### Test scenario(s) for any Magento Backend Supported Configurations





#### Is Browser/Device testing needed?





#### Any ad-hoc/edge case scenarios that need to be considered?




## Screenshots / Screen Captures (if appropriate)

## Breaking Changes (if any)





## Checklist



-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.
